File: GCD_LCM/gcd_lcm.cpp
#include<iostream>
using namespace std;

int gcd(int a,int b){
    if(b == 0){
        return a;
    }
    return gcd(b,a%b);
}

int lcd(int a,int b){
    return (a*b)/gcd(a,b);
}


int main(){

    cout<<gcd(12,18)<<"\n";
    cout<<lcd(12,18);
    return 0;

}

File: GCD_LCM/gcd_lcm.java
import java.util.*;

public class gcd_lcm {
    
    public int gcd(int a,int b){
        if(b==0){
            return a;
        }
        return gcd(b,a%b);
    }
    
    public int lcm(int a,int b){
        return (a*b)/gcd(a, b);
    }

    public static void main(String[] args) {
        gcd_lcm a = new gcd_lcm();
        System.out.println(a.gcd(12,18));
    }
}